### PR TITLE
Feature/dialogue visualisation

### PR DESCRIPTION
**Pull request by davidsalsa**

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 1]
```diff
@@ -26,19 +26,28 @@ public class AdventurerAI : MonoBehaviour
...
+    public float backgroundDialogueHeight = 4.0f;
     public float addedStoppingDistance = 2.0f;
     public GameObject orderPopup;
+    public GameObject speechPopup;
     public Transform wanderpoint;
     public float wanderPointRadius;
+    public float dialogueDuration;
+    public float enteringDialogueChance = 30;
+    public float randomDialogueChance = 30;
+    public bool isSpeaking;
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: This can be private?

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 2]
```diff
@@ -26,19 +26,28 @@ public class AdventurerAI : MonoBehaviour
...
+    private bool hasEntered;
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: _hasEntered

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs (hunk 1 above)
Comment (Lunium) [reply behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: if so _isSpeaking

File: behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        _text = transform.Find("Text").gameObject;
+        _cameras = FindObjectsOfType<Camera>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        foreach (Camera camera in _cameras)
+        {
+            if (camera.CompareTag("FP") && camera.enabled)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs]: There should only be one active camera.
Even if there isn't, you only want to run this code once.
After changing the transform, add a return statement to stop looping the cameras

File: behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                transform.Rotate(Vector3.up, _invertText);
+            }
+            else if (!camera.CompareTag("FP") && camera.enabled)
+            {
+                transform.rotation = camera.transform.rotation;
+            }
+        }
+    }
+
+    public void ResetTimer()
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs]: Remove useless method

File: behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    public void SetContent(string content)
+    {
+        SetVisible(false);
+        _text.GetComponent<TextMeshProUGUI>().text = content;
+        SetVisible(true);
+    }
+
+    public void SetDuration(float timerDuration)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs]: Remove useless method

File: behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        throw new System.NotImplementedException();
+    }
+
+    public void SetPosition(Vector3 position)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/SpeechPopup.cs]: Remove unused method

**Pull request merged**